### ISSUE TITLE
signing_policy parser does not handle escaping

### ISSUE THREAD

**Issue by paulmillar** [eu-emi/canl-java#95]
The [OSG CA distribution v1.72](http://hcc-mirror.unl.edu/osg/3.4/el7/release/x86_64/osg-ca-certs-1.72-1.osg34.el7.noarch.rpm) includes various references to Let's Encrypt CA; e.g., letsencryptauthorityx3.signing_policy

```
access_id_CA   X509    '/C=US/O=Let\'s Encrypt/CN=Let\'s Encrypt Authority X3'
pos_rights     globus  CA:sign
cond_subjects  globus  '"/CN=*"'
```

Note the apostrophes in the X.509 Certificate DN (`/C=US/O=Let's Encrypt/CN=Let's Encrypt Authority X3`).  These apostrophes must be escaped in the signing_policy file (e.g., `../CN=Let\'s Enc...`)

Currently CaNL library does not handle these escaped values correctly.  It yields the following error:

```
Error loading EACL namespace
(signing_policy) from /etc/grid-security/certificates/4f06f81d.signing_policy:
Syntax problem, garbage at the end of line: s Encrypt/CN=Let\'s
EncryptAuthority
X3'
```

This looks like it (incorrectly) accepts the escaped apostrophe as the end of the string, and the remaining string is then interpreted as garbage.

**Comment by paulmillar** [comment]
Resolving this issue is now urgent: production dCache instances are attempting to work with Let's Encrypt issued certificate, which this bug is blocking.

**Comment by golbi** [comment]
@paulmillar ok, I'll find a moment to fix this. I don't have any formal spec of Globus signing policy file, so I'm assuming that \ escaping should work in the access_id_CA entry in a usual way.

**Comment by paulmillar** [comment]
Would it be possible to back-port the fix to the 2.5-branch?

We are currently using 2.5.0 in dCache supported branches, so would appreciate a bug-fix release on that branch.

**Comment by golbi** [comment]
Hmm, if you want to create a backported version then should be rather possible to cherry pick that change to 2.5 fork.
However maybe upgrading to 2.6 would be just fine? BC dependency is upgraded to latest but I've checked that all versions from 1.59 are fine (earlier maybe too - haven't checked). Besides there are 2 bugixes (including this one) and change of default for proxy certificates bit length.

**Comment by paulmillar** [comment]
Those are actually good reasons why we would prefer to have a back-port!

We are currently using BC 1.54, so it's possible upgrading CaNL to 2.6.0 would break something.  The change in cert length (which, in general, is a good idea) could also break something in a client somewhere. IMHO, these changes are too large to be included in a bug-fix release of dCache.

The way we manage such changes (and their risk) is by including them with a new major version of dCache.  In effect, commit the change only to our 'master' branch and not back-porting to the supported branches.  This way we can advertise the changes as part of the release notes for that new major dCache version, and sites have a chance to test the upgrade in their pre-production instances.

This bug affects only one (possible two) sites.  If we were to back-port an upgrade to 2.6.0 as a bug-fix release of dCache then all dCache sites would incur the risks associated with a BC version upgrade (or the risk from not doing the BC version upgrade), changing the default proxy bit-length, etc.

On a practical level, I see this repo currently has no 2.5 branch.  Of course we can create such a branch in our private fork of canl-java and support 2.5 branch ourselves, but we would prefer that this somehow happens within the official CaNL repo, so we don't duplicate effort with other projects that also depend on CaNL.

**Comment by golbi** [comment]
@paulmillar OK, 2.5.1 was pushed should be available soon on meven repo. Includes this fix merged on top of 2.5.0 release tag. 
For subsequent updates please provide pull requests. Branch is named 2.5.x
